src/assets/assetMetrics.js

```diff
@@ -0,0 +1,262 @@
+    };
+    const requiredFields = [
+      'description',
+      'label',
+      'organizationId',
+      'timeInterval'
+    ];
+
+    for (let i = 0; i < requiredFields.length; i++) {
+      const field = requiredFields[i];
+      const hasField = hasFieldFns[field] || hasFieldFns.default;
+
+      if (!hasField(assetMetric, field)) {
+        return Promise.reject(
+          new Error(`A ${field} is required to create a new asset metric.`)
+        );
+      }
+    }
+
+    return this._request
+      .post(
+        `${this._baseUrl}/assets/types/${assetTypeId}/metrics`,
+        toSnakeCase(assetMetric)
+      )
+      .then((assetMetric) => toCamelCase(assetMetric));
+  }
+
+  /**
+   * Deletes an asset metric
+   *
+   * API Endpoint: '/assets/metrics/:assetMetricId'
+   * Method: DELETE
+   *
+   * @param {string} assetMetricId The UUID formatted ID of the asset metric
+   *
+   * @returns {Promise}
+   * @fulfill {undefined}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.assets.metrics.delete('4f0e51c6-728b-4892-9863-6d002e61204d')
+   */
+  delete(assetMetricId) {
+    if (!assetMetricId) {
+      return Promise.reject(
+        new Error(
+          'An asset metric ID is required for deleting an asset metric.'
+        )
+      );
+    }
+
+    return this._request.delete(
+      `${this._baseUrl}/assets/metrics/${assetMetricId}`
+    );
+  }
+
+  /**
+   * Gets information about an asset metric
+   *
+   * API Endpoint: '/assets/metrics/:assetMetricId'
+   * Method: GET
+   *
+   * @param {string} assetMetricId The UUID formatted ID of the asset metric
+   *
+   * @returns {Promise}
+   * @fulfill {AssetMetric} Information about the asset metric
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.assets.metrics
+   *   .get('4f0e51c6-728b-4892-9863-6d002e61204d')
+   *   .then((assetMetric) => console.log(assetMetric))
+   *   .catch((err) => console.log(err));
+   */
+  get(assetMetricId) {
+    if (!assetMetricId) {
+      return Promise.reject(
+        new Error(
+          'An asset metric ID is required for getting information about an asset metric.'
+        )
+      );
+    }
+
+    return this._request
+      .get(`${this._baseUrl}/assets/metrics/${assetMetricId}`)
+      .then((assetMetric) => toCamelCase(assetMetric));
+  }
+
+  /**
+   * Gets a list of all asset metrics that belong to a given type
+   *
+   * API Endpoint: '/assets/types/:assetTypeId/metrics
+   * Method: GET
+   *
+   * @param {string} assetTypeId The UUID formatted ID of the asset type
+   * @param {PaginationOptions} [paginationOptions]
+   *
+   * @returns {Promise}
+   * @fulfill {AssetMetricsFromServer}
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.assets.metrics
+   *   .getAll('4f0e51c6-728b-4892-9863-6d002e61204d')
+   *   .then((assetMetrics) => console.log(assetMetrics))
+   *   .catch((err) => console.log(err));
+   */
+  getAll(assetTypeId, paginationOptions) {
```

Comment:
This doesn't actually get all, so would be better as like `getByAssetTypeId`

Reply:
That sounds good!